### PR TITLE
fix(bake): let's not keep rebaking forever.

### PR DESCRIPTION
**Pull request by emjburns**
Rebaking is extremely unlikely to fix any bake problems. Let's not rebake unless the diff changes.

This pr uses some unhappy controls that asher put into place.

Alternatively to not rebaking, I could change this to rebake every day or every 6 hours instead. Thoughts?

### CODE REVIEW

**Comment by lorin** [comment]
What happens if the user wants to manually attempt a rebake? (e.g., they look at the bake error, it looks to them like a transient issue). How would they do that?

File: keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/api/ImageSpec.kt [hunk 1]
```diff
@@ -12,7 +14,14 @@ data class ImageSpec(
   val storeType: StoreType,
   val artifactStatuses: Set<ArtifactStatus> = enumValues<ArtifactStatus>().toSet(),
   override val application: String // the application an image is baked in
-) : ResourceSpec {
+) : ResourceSpec, UnhappyControl {
   @JsonIgnore
   override val id: String = artifactName
+
+  @JsonIgnore
+  override val maxDiffCount = 2
```
Comment (lorin) [inline keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/api/ImageSpec.kt]: Does this mean it will retry baking twice if it fails?

**Comment by emjburns** [comment]
@lorin currently there's no way to rebake except changing the diff - so you'd need to push a new artifact version to rebake. Or, you could create a pipeline to rebake for you. That's a shitty experience. I'm not convinced it's shittier than randomly seeing 1000 bake tasks cause we're doing something wrong on our end. Thoughts?

File: keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/api/ImageSpec.kt (hunk 1 above)
Comment (emjburns) [reply keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/api/ImageSpec.kt]: Yes it does I think. I just stole whatever number we were using for clusters w/o digging too much into it

**Comment by lorin** [comment]
@emjburns My first instinct is to provide some sort of UI to allow the user to manually override an "UnhappyVeto" state. That would provide a way for them to try the bake again in this case. (I wouldn't gate this PR on that, although I'd like to know how to manually do that behind the scenes by tweaking the database if somebody asked for this in a support context).

**Comment by emjburns** [comment]
We have the capability to do that! you can "pass a message" to a veto (bad naming...it's my fault). So, we could theoretically have an api you could hit to trigger a rebake. We'd have to make some changes in `UnhappyVeto` to support that.

As far as what to do to unblock a user, i'm pretty sure you could just delete the right row from the `unhappy_veto` table

File: keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/api/ImageSpec.kt [hunk 2]
```diff
@@ -12,7 +14,14 @@ data class ImageSpec(
...
+
+  // rebaking does not help remedy any of the bake problems we see.
```
Comment (luispollo) [inline keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/api/ImageSpec.kt]: So zero means "wait forever or until the diff changes". I feel the comment doesn't quite fit with that. Maybe just remove the comment?

File: keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/api/ImageSpec.kt (hunk 1 above)
Comment (asher) [reply keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/api/ImageSpec.kt]: It's the number of attempts in total, so 1 retry.

Comment (robfletcher) [reply keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/api/ImageSpec.kt]: That seems to conflict with what the PR says which is that there's no point retrying. Shouldn't this be `1`?

Comment (emjburns) [reply keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/api/ImageSpec.kt]: Ehhh, maybe i'll leave it at 2 so it retries once. I can always bump it down.

File: keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/api/ImageSpec.kt (hunk 2 above)
Comment (emjburns) [reply keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/api/ImageSpec.kt]: Sure, I"ll remove it.

**Comment by emjburns** [comment]
added an issue for a rebake api

**Pull request merged**